File: melant_ia/src/walkie_talkie.js
// walkie_talkie.js
// Módulo base para Walkie Talkie MELANTIA (Web/PWA)
class MensajeTexto {
  constructor(texto, autor, timestamp = Date.now()) {
    this.texto = texto;
    this.autor = autor;
    this.timestamp = timestamp;
  }
}

class MensajeAudio {
  constructor(blobUrl, autor, timestamp = Date.now()) {
    this.blobUrl = blobUrl;
    this.autor = autor;
    this.timestamp = timestamp;
    this.escuchado = false;
    this.timestampEscucha = null;
  }
}

class WalkieTalkie {
  constructor(maxMensajes = 10) {
    this.mensajes = [];
    this.maxMensajes = maxMensajes;
    this.audios = [];
  }

  agregarMensaje(texto, autor) {
    if (this.mensajes.length >= this.maxMensajes) {
      this.mensajes.shift(); // Elimina el más antiguo
    }
    this.mensajes.push(new MensajeTexto(texto, autor));
  }

  agregarAudio(blobUrl, autor) {
    const audio = new MensajeAudio(blobUrl, autor);
    this.audios.push(audio);
    return audio;
  }

  marcarAudioEscuchado(audio) {
    audio.escuchado = true;
    audio.timestampEscucha = Date.now();
  }

  limpiarAudiosViejos(segundos = 30) {
    const ahora = Date.now();
    this.audios = this.audios.filter(
      (a) => !(a.escuchado && ahora - a.timestampEscucha > segundos * 1000)
    );
  }

  obtenerMensajes() {
    return [...this.mensajes];
  }

  obtenerAudios() {
    return [...this.audios];
  }
}

File: melant_ia/src/walkie_talkie.py
# walkie_talkie.py
# Módulo base para Walkie Talkie MELANTIA (Kivy/Python)
from collections import deque
import time

class MensajeTexto:
    def __init__(self, texto, autor, timestamp=None):
        self.texto = texto
        self.autor = autor
        self.timestamp = timestamp or time.time()

class MensajeAudio:
    def __init__(self, ruta_audio, autor, timestamp=None, escuchado=False):
        self.ruta_audio = ruta_audio
        self.autor = autor
        self.timestamp = timestamp or time.time()
        self.escuchado = escuchado
        self.timestamp_escucha = None

class WalkieTalkie:
    def __init__(self, max_mensajes=10):
        self.mensajes = deque(maxlen=max_mensajes)  # FIFO mensajes texto
        self.audios = []  # Lista de MensajeAudio

    def agregar_mensaje(self, texto, autor):
        self.mensajes.append(MensajeTexto(texto, autor))

    def agregar_audio(self, ruta_audio, autor):
        audio = MensajeAudio(ruta_audio, autor)
        self.audios.append(audio)
        return audio

    def marcar_audio_escuchado(self, audio):
        audio.escuchado = True
        audio.timestamp_escucha = time.time()

    def limpiar_audios_viejos(self, segundos=30):
        ahora = time.time()
        self.audios = [a for a in self.audios if not (a.escuchado and (ahora - a.timestamp_escucha > segundos))]

    def obtener_mensajes(self):
        return list(self.mensajes)

    def obtener_audios(self):
        return list(self.audios)
